packages/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -199,7 +201,7 @@ export function useGridDimensions(
         height: rootDimensionsRef.current.height,
       };
       viewportInnerSize = {
-        width: Math.max(0, viewportOuterSize.width - leftPinnedWidth - rightPinnedWidth),
+        width: Math.max(0, viewportOuterSize.width),
```

Comment:
IIUC, we're changing the meaning of `viewportInnerSize` to now include pinned columns?